### ISSUE TITLE
[Question] Non-Link usage

### ISSUE THREAD

**Issue by Noitidart** [tatethurston/nextjs-routes#36]
I wanted to use this in a non-link. For example a fetch like `fetch('/foo')`, is there anyway to typecheck that foo?

I was thinking of doing a type like:

```
const ResourcePath = Parameters<typeof useRouter>[0]['path'];
const path: ResourcePath = '/foo';
fetch(path);
```

But it's kind of extra to create a variable to hold path then type that.

**Comment by tatethurston** [comment]
Hey @Noitidart,

You can [import Route directly](https://github.com/tatethurston/nextjs-routes#route).

So you would have:

```ts
import type { Route } from "nextjs-routes";

const path: Route["pathname"] = '/foo';
fetch(path);
```

If you want to remove the variable, you can create a wrapper for fetch:

```ts
function fetchPage(url: Route): ReturnType<typeof fetch> {
  return fetch(url);
}
```

This will work for static pathnames, but `fetch` won't interpolate Nextjs query parameters (eg `/foos/[foo]`).

**Comment by Noitidart** [comment]
Thanks very much for your detailed reply. I wanted to use this for my api routes as well. For pages I would use useRouter.push thanks so much for this awesome lib!

**Comment by tatethurston** [comment]
Ah got it, I’ll think on the API use case some more. There’s certainly an opportunity for a better typescript story there. Thanks @Noitidart !

**Comment by Noitidart** [comment]
Thanks very much sir!

**Comment by tatethurston** [comment]
@Noitidart what do you think about the following API?

```diff
+ import { route } from 'nextjs-routes';

- const ResourcePath = Parameters<typeof useRouter>[0]['path'];
- const path: ResourcePath = '/foo';
- fetch(path);
+ fetch(route({ pathname: '/foo' });
```

`route` would be a runtime export from nextjs-routes that performs the same dynamic query interpolation as Next's link, so  `route({ pathname: '/foos/[foo]', query: { foo: 'bar' }})` would yield `/foos/bar`.

**Comment by Noitidart** [comment]
Woow! That's amazing! Thanks very much for asking!

**Comment by tatethurston** [comment]
This is released in [v0.0.21](https://github.com/tatethurston/nextjs-routes/releases/tag/v0.0.21)

**Comment by Noitidart** [comment]
Thank you man!!
